src/pke/lib/scheme/ckksrns/ckksrns-parametergeneration.cpp

```diff
@@ -152,20 +150,21 @@ bool ParameterGenerationCKKSRNS::ParamsGenCKKSRNS(std::shared_ptr<CryptoParamete
        * scale factor of level 0 as possible.
        */
 
-            double sf    = moduliQ[numPrimes - 1].ConvertToDouble();
-            uint32_t cnt = 0;
-            for (usint i = numPrimes - 2; i >= 1; i--) {
-                sf = static_cast<double>(pow(sf, 2) / moduliQ[i + 1].ConvertToDouble());
+            // TODO (dsuponit): do we need to initialize sf here at all? seems like it should be moved inside the loop
```

Comment:
Yes, because we're squaring it each time in the loop. We need some initial value.